### PR TITLE
Improve syntaxTreeModify to remove imports automatically

### PR DESCRIPTION
**Pull request by suhothayan**
## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ x] Added necessary tests
  - [x ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorListener.java [hunk 1]
```diff
@@ -47,7 +45,8 @@ public void reportInvalidNodeError(STToken token, String message) {
     }
     
     private void logError(int line, int col, String message) {
-        PrintStream out = System.out;
-        out.println("xxx.bal:" + line + ":" + col + ":" + message);
+        // TODO - Removing below lines as it breaks lang-server rpc
```
Comment (sameerajayasoma) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorListener.java:48]: We are fixing this with the following PR. 

https://github.com/ballerina-platform/ballerina-lang/pull/23500

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/incremental/HybridNodeStorage.java [hunk 2]
```diff
@@ -80,6 +80,9 @@ HybridNode peekSubtree() {
     }
 
     HybridNode consumeToken() {
+        if (currentToken == null) {
+            currentToken = peekToken();
+        }
         HybridNode hybridNode = currentToken;
```
Comment (sameerajayasoma) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/incremental/HybridNodeStorage.java:86]: Can you please create an issue for this change? I think we need to fix this in other places too. Issue will help us to track it. Thanks.

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/tree/STNode.java [hunk 3]
```diff
@@ -156,7 +156,7 @@ private void updateWidth(STNode[] children, int firstChildIndex, int lastChildIn
     private int getFirstChildIndex(STNode... children) {
         for (int index = 0; index < children.length; index++) {
             STNode child = children[index];
-            if (SyntaxUtils.isSTNodePresent(child) && child.widthWithMinutiae != 0) {
+            if (SyntaxUtils.isSTNodePresent(child) && child.widthWithLeadingMinutiae != 0) {
```
Comment (sameerajayasoma) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/tree/STNode.java:159]: Can you please create an issue explaining the bug here?

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/text/StringTextDocument.java [hunk 4]
```diff
@@ -28,7 +28,7 @@
     private final String text;
     private LineMap textLineMap;
 
-    StringTextDocument(String text) {
+    public StringTextDocument(String text) {
```
Comment (sameerajayasoma) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/text/StringTextDocument.java:31]: This class is not part of the public syntax API. I think I've mistakenly made this class public. 

Why do you want to directly access this class anyway?

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorListener.java (hunk 1 above)
Comment (suhothayan) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorListener.java:48]: Noted

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/incremental/HybridNodeStorage.java (hunk 2 above)
Comment (suhothayan) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/incremental/HybridNodeStorage.java:86]: Created an issue https://github.com/ballerina-platform/ballerina-lang/issues/23719

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/tree/STNode.java (hunk 3 above)
Comment (suhothayan) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/tree/STNode.java:185]: This seems to be unnecessary now. I'll revert this.

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/text/StringTextDocument.java (hunk 4 above)
Comment (suhothayan) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/text/StringTextDocument.java]: This is not needed, as there are other ways to create StringTextDocument. I'll revert this.

**Pull request merged**